examples/accumulators/template_function_accumulator.hpp

```diff
@@ -106,23 +106,25 @@ namespace examples
         ///          the future should be called. If the value is available,
         ///          get() will return immediately; otherwise, it will block
         ///          until the value is ready.
-        hpx::future<double> query_async()
+        hpx::future<double> query(hpx::launch::async_policy)
         {
             HPX_ASSERT(this->get_id());
 
             typedef server::template_function_accumulator::query_action
                 action_type;
-            return hpx::async<action_type>(this->get_id());
+            return hpx::async<action_type>(hpx::launch::async, this->get_id());
         }
 
         /// Query the current value of the accumulator.
         ///
         /// \note This function is fully synchronous.
-        double query_sync()
+        double query(hpx::launch::sync_policy = hpx::launch::sync)
         {
             HPX_ASSERT(this->get_id());
 
-            return query_async().get();
+            typedef server::template_function_accumulator::query_action
+                action_type;
+            return action_type()(hpx::launch::sync, this->get_id());
```

Comment:
Not quite. `action_type()(...)` was always equivalent to normal function invocation, correct. However, what I meant was this:
```
future<R> f = action()(hpx::launch::async, ....);
R r = action()(hpx::launch::sync, ...); // this is what we currently have.
```